shield.go

```diff
@@ -9,27 +9,41 @@ import (
 	"golang.org/x/crypto/pbkdf2"
 )
 
-// Iterations is the number of iterations to run PBKDF2.
-const Iterations = 5000
-
-// MaxLength indicates the maximum length that a password can have.
-const MaxLength = 4096
+var ErrPasswordTooLong = errors.New("password too long")
 
-// SaltLength is the length of the salt generated by GenerateSalt.
-const SaltLength = 32
+// Shield is a struct that contains basic configuration variables used to
+// encrypt and check strings.
+//     iterations: is the number of iterations to run PBKDF2.
+//     maxLength: indicates the maximum length that a password can have.
+//     saltLength: is the length of the salt generated by GenerateSalt.
+type Shield struct {
+	iterations int
+	maxLength  int
```

Comment:
I'm not sure this should be a configurable value, but a constant instead. `4096` was chosen as a constant because, as it [says in `Shield::Password` comments](https://github.com/cyx/shield/blob/master/lib/shield.rb#L103-L109): a very long password would hang a server running this operation, thus DoS'ing it. Check the [referenced blog post](https://www.djangoproject.com/weblog/2013/sep/15/security/) as well.